src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -257,3 +258,11 @@ export const extractRegisteredAuthTypeCredentials = (
 
   return registeredCredentials;
 };
+
+let application: ApplicationStart;
```

Comment:
Just want to be clear, can we avoid having mutable global variables?

Reply:
sure, moving it to a function to encapsulate, and use singleton pattern